tests/func/test_checkout.py

```diff
@@ -880,3 +880,31 @@ def test_checkout_executable(tmp_dir, dvc):
         assert not isexec
     else:
         assert isexec
+
+
+def test_checkout_partial(tmp_dir, dvc):
+    tmp_dir.dvc_gen(
+        {"db": {"a.txt": "a", "b.txt": "b", "sub_dir": {"c.txt": "c"}}}
+    )
+
+    db_dir = tmp_dir / "db"
+    sub_dir = db_dir / "sub_dir"
+    file_1, file_2, sub_dir_file = [
```

Comment:
Should we also add a test for `sub_dir`?

Reply:
Can you be more explicit about the case? We already have a test case to checkout a partial file from a subdirectory but if you mean anything, we can add that too!

Reply:
@isidentical I believe @skshetry is saying that we should test `dvc checkout db/sub_dir` too. I left a comment about it as well.

Reply:
Oh, right! Sure, can add it.